Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -48,41 +64,82 @@ void APSStarActor::OnLocalCharacterReady(APlayerCharacter* Character, int32 Char
 	}
 }
 
+// Called when the current game state was changed
+void APSStarActor::OnGameStateChanged(ECurrentGameState GameState)
+{
+	if (AMyGameStateBase::GetCurrentGameState() == ECurrentGameState::Menu)
+	{
+		SetStartTimeMenuStars();
+		TryPlayMenuStarAnimation();
+	}
+	else
+	{
+		StartTimeMenuStarsInternal = 0.f;
+	}
+}
+
 // Is called when any cinematic started
 void APSStarActor::OnAnyCinematicStarted(const UObject* LevelSequence, const UObject* FromInstigator)
 {
-	const UWorld* World = GetWorld();
-	check(World);
-
-	StartTimeHideStarsInternal = World->GetTimeSeconds();
-
-	// Enable tick and process this frame
-	SetActorTickEnabled(true);
+	SetStartTimeHideStars();
 	TryPlayHideStarAnimation();
 }
 
-// Hiding stars with animation
+// Hiding stars with animation in main menu when cinematic is start to play
 void APSStarActor::TryPlayHideStarAnimation()
 {
-	if (!StartTimeHideStarsInternal)
+	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
+	const bool bIsFinished = !TryPlayStarAnimation(StartTimeHideStarsInternal, CurrentRow.HideStarsAnimation);
+	if (bIsFinished)
 	{
-		return;
+		StartTimeHideStarsInternal = 0.f;
+		UPoolManagerSubsystem::Get().ReturnToPool(this);
 	}
+}
 
+// Menu stars with animation in main menu idle 
+void APSStarActor::TryPlayMenuStarAnimation()
+{
 	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
-	if (!CurrentRow.HideStarsAnimation)
+	bool bIsFinished = !TryPlayStarAnimation(StartTimeMenuStarsInternal, CurrentRow.MenuStarsAnimation);
```

Comment:
Looks like the `const-police` is on the case! 🚨
You’ve got a variable that’s not supposed to change, but it's missing its `const` badge.

Please always remember to add `const` where possible:
```cpp
const bool bIsFinished = //...
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/1ee3a71dc961083730f10984bda49e05a658f2e8